x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CacheFile.java

```diff
@@ -398,4 +406,52 @@ private FileChannelReference acquireFileChannelReference() {
         ensureOpen();
         return tracker.getAbsentRangeWithin(start, end);
     }
+
+    // used in tests
+    boolean isFSynced() {
+        return fsynced.get();
+    }
+
+    /**
+     * Ensure that all ranges of data written to the cache file are written to the storage device that contains it. This method returns the
+     * list of all successfully written ranges of data since the creation of the cache file.
+     *
+     * @return the list of ranges of data available in cache at the time this method is invoked
+     * @throws IOException                       if the cache file failed to be fsync
+     * @throws AlreadyClosedException            if the cache file is evicted
+     * @throws java.nio.file.NoSuchFileException if the cache file does not exist
+     */
+    public List<Tuple<Long, Long>> fsync() throws IOException {
```

Comment:
Should this return a `SortedSet` instead to signal the expected order verified by the test, as well as uniqueness?

Reply:
This is a good suggestion, thanks. I pushed 7b3590edaa921b328efb5bd5c97162f9795b1c55